### ISSUE TITLE
Issue (sync_watch_status)

### ISSUE THREAD

**Issue by ReenigneArcher** [blacktwin/JBOPS#249]
**Describe the bug**
Script stopped working (not sure when). Tried in python 2 and 3.

Python 2 called with:
`py -2 sync_watch_status.py --userFrom ReenigneArcher=Archer --userTo Dave=Archer --libraries "Movies" "TV Shows" "Sports" "TV Sports" "YouTube Videos"`

_Python 2 output:_
```
Connecting to admin server(s) for access info...
Checking ReenigneArcher's access to Archer
Connecting ReenigneArcher to Archer...
Traceback (most recent call last):
  File "sync_watch_status.py", line 472, in <module>
    watchedFrom = check_users_access(plex_access, userFrom, serverFrom, libraries)
  File "sync_watch_status.py", line 346, in check_users_access
    server_connection = connect_to_server(server_obj, _user['account'])
  File "sync_watch_status.py", line 314, in connect_to_server
    user_connection = Plex(url=url, token=token)
  File "sync_watch_status.py", line 198, in __init__
    self.server = PlexServer(baseurl=url, token=token, session=session)
  File "C:\Python27\lib\site-packages\plexapi\server.py", line 102, in __init__
    data = self.query(self.key, timeout=timeout)
  File "C:\Python27\lib\site-packages\plexapi\server.py", line 343, in query
    response = method(url, headers=headers, timeout=timeout, **kwargs)
  File "C:\Python27\lib\site-packages\requests\sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "C:\Python27\lib\site-packages\requests\sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Python27\lib\site-packages\requests\sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "C:\Python27\lib\site-packages\requests\adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='45.79.222.231', port=8443): Max retries exceeded with url: / (Caused by SSLError(SSLEOFError(8, u'EOF occurred in violation of protocol (_ssl.c:727)'),))
```

Python 3 called with:
`py -3 sync_watch_status.py --userFrom ReenigneArcher=Archer --userTo Dave=Archer --libraries "Movies" "TV Shows" "Sports" "TV Sports" "YouTube Videos"`

_Python 3 output:_
```
e:\Scripts\JBOPS-master\utility>py -3 sync_watch_status.py --userFrom ReenigneArcher=Archer --userTo Dave=Archer --libraries "Movies" "TV Shows" "Sports" "TV Sports" "YouTube Videos"
Connecting to admin server(s) for access info...
Checking ReenigneArcher's access to Archer
Connecting ReenigneArcher to Archer...
Traceback (most recent call last):
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 978, in _validate_conn
    conn.connect()
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connection.py", line 362, in connect
    self.sock = ssl_wrap_socket(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\util\ssl_.py", line 397, in ssl_wrap_socket
    return context.wrap_socket(sock)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\ssl.py", line 1040, in _create
    self.do_handshake()
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\adapters.py", line 439, in send
    resp = conn.urlopen(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 754, in urlopen
    return self.urlopen(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 754, in urlopen
    return self.urlopen(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 754, in urlopen
    return self.urlopen(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 726, in urlopen
    retries = retries.increment(
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\util\retry.py", line 439, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='23.92.30.53', port=8443): Max retries exceeded with url: / (Caused by ProtocolError('Connection aborted.', FileNotFoundError(2, 'No such file or directory')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "sync_watch_status.py", line 472, in <module>
    watchedFrom = check_users_access(plex_access, userFrom, serverFrom, libraries)
  File "sync_watch_status.py", line 346, in check_users_access
    server_connection = connect_to_server(server_obj, _user['account'])
  File "sync_watch_status.py", line 314, in connect_to_server
    user_connection = Plex(url=url, token=token)
  File "sync_watch_status.py", line 198, in __init__
    self.server = PlexServer(baseurl=url, token=token, session=session)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\plexapi\server.py", line 110, in __init__
    data = self.query(self.key, timeout=timeout)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\plexapi\server.py", line 440, in query
    response = method(url, headers=headers, timeout=timeout, **kwargs)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "C:\Users\archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='23.92.30.53', port=8443): Max retries exceeded with url: / (Caused by ProtocolError('Connection aborted.', FileNotFoundError(2, 'No such file or directory')))
```

**Link to script with bug/issue**
[sync_watch_status.py](https://github.com/blacktwin/JBOPS/blob/master/utility/sync_watch_status.py)

**To Reproduce**
1. Not sure how to reproduce. I am using the config.ini with myplex_username, myplex_password, server_baseurl (http://localhost:32400), and server_token. I have an environment variable PLEXAPI_CONFIG_PATH pointing to the config file. None of this has changed.

**Expected behavior**
Just want the script to run as before.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Win 10 Pro
 - Browser: Firefox
 - Python Version: 2.7.16 and 3.8.5

**Additional context**

**Comment by blacktwin** [comment]
Just ran a test on my for 2.7 and 3.8 and both worked without issue. 

Try editing these variables in the script:
```
PLEX_TOKEN = ''
TAUTULLI_URL = ''
TAUTULLI_APIKEY = ''
```
That will override the use of the config file. Your `PLEX_TOKEN` in the config file maybe bad?

**Comment by ReenigneArcher** [comment]
Same issue if I put the token directly in the script. I double checked the token from the url of a movie's metadata xml. I'm running this outside of Tautulli for now.

I think the issue is somehow related to app.plex.tv. For some reason since I made my server accessible from my own domain name, when I log into app.plex.tv my server can't be found if I'm not on my local network. If I'm on the local network it only does an indirect connection.

If I connect to my domain though everything works perfectly.

Anyway I never had the pkcs12 certificate in my plex settings. I added the pfx archive there last night along with the password used when making the pfx. I still can't connect using app.plex.tv but I get a little further in the script now, although still it won't login. This is my output now using python 3.

`Traceback (most recent call last):
  File "sync_watch_status.py", line 437, in <module>
    plex_admin = Plex(PLEX_TOKEN)
  File "sync_watch_status.py", line 198, in __init__
    self.account = MyPlexAccount(token)
  File "C:\Users\Arhcer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\plexapi\myplex.py", line 94, in __init__
    data, initpath = self._signin(username, password, timeout)
  File "C:\Users\Arhcer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\plexapi\myplex.py", line 102, in _signin
    data = self.query(self.SIGNIN, method=self._session.post, auth=(username, password), timeout=timeout)
  File "C:\Users\Arhcer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\plexapi\myplex.py", line 191, in query
    raise Unauthorized(message)
plexapi.exceptions.Unauthorized: (401) unauthorized; https://plex.tv/users/sign_in.xml <?xml version="1.0" encoding="UTF-8"?> <errors>   <error>Invalid email, username, or password.</error> </errors>
`

I'm running the script from the plex server itself. Is there a way to bypass it from using app.plex.tv? I'm really not sure why I can't connect using app.plex.tv anymore. Thought I set all that up correctly.

Remote access says it's fully accessible outside my network. In the network settings this is what I have.
Secure connections: Preferred
Custom certificate location: _full_path_to_my_certificate_.pfx
Custom certificate encryption key: _password used when creating the pfx in win-acme_
Custom certificate domain: _mysite_.duckdns.org
Custom server access URLs: _mysite_.duckdns.org
List of IP addresses and networks that are allowed without auth: 192.168.1.1/255.255.255.0

**Comment by blacktwin** [comment]
Based on that error message you are not even able to sign into your Plex account. The script connects to your Plex account first then connects to whatever servers you select. You haven't connected to your Plex account yet.

> Is there a way to bypass it from using app.plex.tv?

Yeah I think so. Change [this](https://github.com/blacktwin/JBOPS/blob/master/utility/sync_watch_status.py#L307-L317):
```python
    print('Connecting {} to {}...'.format(user, server_name))
    server_connection = server_obj.connect()
    baseurl = server_connection._baseurl.split('.')
    url = "".join([baseurl[0].replace('-', '.'),
                   baseurl[-1].replace('direct', '')])
    if user_account.title == Plex(PLEX_TOKEN).account.title:
        token = PLEX_TOKEN
    else:
        token = user_account.get_token(server_connection.machineIdentifier)

    user_connection = Plex(url=url, token=token)
```
to this:
```python
    PLEX_URL = "http://localhost:32400"
    if user_account.title == Plex(PLEX_TOKEN).account.title:
        token = PLEX_TOKEN
    else:
        token = user_account.get_token(server_connection.machineIdentifier)

    user_connection = Plex(url=PLEX_URL, token=token)
```

But you still need to be able to sign into your Plex account firs.t

**Comment by ReenigneArcher** [comment]
Okay... made a little progress, still not working though.

I reset my plex password, and got a new server token. This had no effect.

I then decided to test the remote access setting. I set it to manual instead of automatic. For some reason UPnP, while working according to the remote access settings, didn't allow the connection. I can now connect using app.plex.tv and I have a proper connection!

But with the script/plexapi still now working.
Now getting some new error about a "new connection" with Python 3 (using original script).

_Here's the output_
Connecting to admin server(s) for access info...
Checking ReenigneArcher's access to Archer
Connecting ReenigneArcher to Archer...
Traceback (most recent call last):
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connection.py", line 159, in _new_conn
    conn = connection.create_connection(
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\util\connection.py", line 61, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 11001] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 978, in _validate_conn
    conn.connect()
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connection.py", line 309, in connect
    conn = self._new_conn()
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connection.py", line 171, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x03C3A9D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\adapters.py", line 439, in send
    resp = conn.urlopen(
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 754, in urlopen
    return self.urlopen(
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 754, in urlopen
    return self.urlopen(
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 754, in urlopen
    return self.urlopen(
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\connectionpool.py", line 726, in urlopen
    retries = retries.increment(
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\urllib3\util\retry.py", line 439, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='archer.plexorg', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x03C3A9D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "sync_watch_status.py", line 475, in <module>
    watchedFrom = check_users_access(plex_access, userFrom, serverFrom, libraries)
  File "sync_watch_status.py", line 349, in check_users_access
    server_connection = connect_to_server(server_obj, _user['account'])
  File "sync_watch_status.py", line 317, in connect_to_server
    user_connection = Plex(url=url, token=token)
  File "sync_watch_status.py", line 201, in __init__
    self.server = PlexServer(baseurl=url, token=token, session=session)
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\plexapi\server.py", line 110, in __init__
    data = self.query(self.key, timeout=timeout)
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\plexapi\server.py", line 440, in query
    response = method(url, headers=headers, timeout=timeout, **kwargs)
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "C:\Users\Archer\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='archer.plexorg', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x03C3A9D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))

**Comment by ReenigneArcher** [comment]
I'm not sure but I think it's trying to go to the wrong url to access my server. Can I have it print the url before the traceback begins?

If the url is archer.plexorg as in the last line of my previous message, that definitely isn't right. I noticed [here ](https://github.com/blacktwin/JBOPS/blob/41ed4199e10f165ae4a2e45cb4e7f3361e72789d/utility/sync_watch_status.py#L310) there is a replace. I do have a "-" in my domain.

**Comment by ReenigneArcher** [comment]
With that I get the following output:

> Connecting to admin server(s) for access info...
> Checking ReenigneArcher's access to Archer
> Archer
> ReenigneArcher
> Finding watched items in libraries...
> Archer
> Dave
> Traceback (most recent call last):
>   File "sync_watch_status2.py", line 480, in <module>
>     plexTo.append([user, check_users_access(plex_access, user, server_name, libraries)])
>   File "sync_watch_status2.py", line 347, in check_users_access
>     server_connection = connect_to_server(server_obj, _user['account'])
>   File "sync_watch_status2.py", line 313, in connect_to_server
>     token = user_account.get_token(server_connection.machineIdentifier)
> NameError: name 'server_connection' is not defined

**Comment by ReenigneArcher** [comment]
Okay this script is working now.

I don't understand what the problem is though.

If I change the "Custom certificate encryption key" setting it works just fine.
This works, but my actual encryption key doesn't work...
![image](https://user-images.githubusercontent.com/42013603/96060124-a6e98000-0e5d-11eb-942e-bad526e8de6c.png)

**Comment by blacktwin** [comment]
I don't use a domain or custom certs so it's probably something that I'm did not caught because my setup is different. I updated the code snippet to resolve the error you saw. Did you make any other edits other than the cert change?

**Comment by ReenigneArcher** [comment]
From my initial problem here's everything I changed.

- added pfx certificate to Plex (I only had it in NGINX proxy manager before)
- added encryption key to Plex (doesn't work when correct key is there, but does work when any random key is there)
- changed Plex from using UPnP to port forwarding for remote access (this allowed me to connect using app.plex.tv)

I think I don't actually need the pfx cert and key in plex as it's handled by NGINX. I've removed those 3 settings from Plex and it appears to still let me connect using app.plex.tv. So the only change I needed was to go from UPnP to port forwarding on my router. Not sure what changed there.
